tests/services/components/test_internal_notes_component.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Tests for the service Metadata component."""
+from copy import deepcopy
+
+from invenio_rdm_records.proxies import current_rdm_records
+from invenio_rdm_records.records import RDMRecord
+from invenio_rdm_records.records.api import RDMDraft
+from invenio_rdm_records.services.components.internal_notes import (
+    InternalNotesComponent,
+)
+
+
+def test_notes_component(minimal_record, parent, identity_simple, location):
+    """Test the metadata component."""
+    record = RDMRecord.create(minimal_record, parent=parent)
+    draft = RDMDraft.new_version(record)
+
+    assert "publication_date" in record.metadata
+    assert "title" in record.metadata
+
+    component = InternalNotesComponent(current_rdm_records.records_service)
+    new_data = deepcopy(draft)
+    new_data["internal_notes"] = [{"note": "abc", "id": "abcd-efgh"}]
+    # check if internal notes have correct attributes
+    component.update_draft(identity_simple, data=new_data, record=draft)
+    assert "internal_notes" in draft
+    notes = draft["internal_notes"]
+    assert "id" in notes[0]
+    assert "timestamp" in notes[0]
+    assert "added_by" in notes[0]
+
+    # check if new internal note is added properly
+    new_data["internal_notes"] = notes + [{"note": "def", "id": "frgt-09df"}]
```

Comment:
Here I think the test is wrong because it assumes that a new note will have an `id` unless I am mistaken